bigquery_etl/glam/templates/probe_counts_v1.sql

```diff
@@ -1,98 +1,130 @@
+      FROM
+        buckets
+      CROSS JOIN
+        UNNEST(arr) AS item
     )
-    SELECT
-      ARRAY_AGG(CAST(item AS INT64))
-    FROM
-      buckets
-    CROSS JOIN
-      UNNEST(arr) AS item
-  )
-);
-{% endif %}
+  );
 
-SELECT
+{% endif %}
+WITH probe_counts AS (
+  SELECT
     {{ attributes }},
     {{ aggregate_attributes }},
     {% if is_scalar %}
-        client_agg_type,
-        agg_type,
-        {%if channel == "release" %}
-          -- Logic to count clients based on sampled windows release data, which started in v119.
-          -- If you're changing this, then you'll also need to change
-          -- clients_daily_[scalar | histogram]_aggregates
-          IF(os = 'Windows' AND app_version >= 119,
-            SUM(count) * 10,
-            SUM(count)
-          ) AS total_users,
-        {% else %}
-          SUM(count) AS total_users,
-        {% endif %}
-        mozfun.glam.histogram_fill_buckets_dirichlet(
-            mozfun.map.sum(ARRAY_AGG(STRUCT<key STRING, value FLOAT64>(bucket, count))),
-            CASE
-              WHEN metric_type IN ({{ scalar_metric_types }})
-                THEN ARRAY(SELECT FORMAT("%.*f", 2, bucket) FROM UNNEST(
-                    mozfun.glam.histogram_generate_scalar_buckets(range_min, range_max, bucket_count)
-                  ) AS bucket ORDER BY bucket)
-              WHEN metric_type in ({{ boolean_metric_types }})
-                THEN ['always', 'never', 'sometimes']
-            END,
-            SUM(count)
-        )
-        AS aggregates
+      client_agg_type,
+      agg_type,
+      {%if channel == "release" %}
+            -- Logic to count clients based on sampled windows release data, which started in v119.
+            -- If you're changing this, then you'll also need to change
+            -- clients_daily_[scalar | histogram]_aggregates
+        IF(os = 'Windows' AND app_version >= 119, SUM(count) * 10, SUM(count)) AS total_users,
+      {% else %}
+        SUM(count) AS total_users,
+      {% endif %}
+      mozfun.glam.histogram_fill_buckets_dirichlet(
+        mozfun.map.sum(ARRAY_AGG(STRUCT<key STRING, value FLOAT64>(bucket, count))),
+        CASE
+          WHEN metric_type IN ({{ scalar_metric_types }})
+            THEN ARRAY(
+                SELECT
+                  FORMAT("%.*f", 2, bucket)
+                FROM
+                  UNNEST(
+                    mozfun.glam.histogram_generate_scalar_buckets(
+                      range_min,
+                      range_max,
+                      bucket_count
+                    )
+                  ) AS bucket
+                ORDER BY
+                  bucket
+              )
+          WHEN metric_type IN ({{ boolean_metric_types }})
+            THEN ['always', 'never', 'sometimes']
+        END,
+        SUM(count)
+      ) AS aggregates
     {% else %}
-        agg_type AS client_agg_type,
-        'histogram' as agg_type,
-        {% if channel == "release" %}
```

Comment:
Is this not needed anymore because the sample value were multiplied by 10 in `histogram_bucket_counts`?

Reply:
That's right. I was able to correct the histogram counts at the `bucket_counts` level, but because scalars use `HLL` I had to fix it at `probe_counts`, which is a step later